iree/tools/BUILD

```diff
@@ -146,12 +146,14 @@ cc_library(
     copts = [
         "-DIREE_HAVE_LLVMAOT_TARGET",
         "-DIREE_HAVE_LLVMIR_TARGET",
+        "-DIREE_HAVE_METAL_TARGET",
         "-DIREE_HAVE_VMLA_TARGET",
         "-DIREE_HAVE_VULKANSPIRV_TARGET",
     ],
     deps = [
         "//iree/compiler/Dialect/HAL/Target/LLVM/AOT:LLVMAOT",
         "//iree/compiler/Dialect/HAL/Target/LLVM/IR:LLVMIR",
+        "//iree/compiler/Dialect/HAL/Target/MetalSPIRV",
```

Comment:
We generally want adding a new platform to not break or complicate existing platforms. I think the Metal compiler target here is fine (assuming it can build/run on non-Apple platforms), but the runtime (HAL) `objc_library`s will definitely need some special handling.

We can add a `select()` here: https://github.com/google/iree/blob/d3e695a0483d3ab3e9dd7c881b6a9c010d288480/iree/build_defs.oss.bzl#L53-L59

but we'll also need ways to filter from `bazel build iree/...` style commands, including what our build bots run (e.g. https://github.com/google/iree/blob/main/build_tools/bazel/build_core.sh), unless that has improved in Bazel since I last looked at it..

Reply:
Yes, `select()` should be used. I'll check what can be done for `bazel build iree/....`.